### PR TITLE
Add unknown primitive type

### PR DESCRIPTION
**Pull request by roironn**

### CODE REVIEW

**Comment by omrilit** [comment]
Not sure whether this type should be `PrimitiveType` or `ObjectType`. I already started to implement a solution with marking it as `ObjectType` and closed it few months ago since it felt pretty dangerous and I could workaround it using a different solution to the problem I had. Please take a look on this PR comments.

**Comment by ori-moisis** [comment]
I think this is what Omri is referring to: https://github.com/salto-io/salto/pull/1247#issuecomment-663968373
while making it a primitive kind can make sense at a high level, it might be safer (and also make sense in other ways) to make it an object type and put it as one of the `BuiltinTypes` (so the special thing about it would be the ID rather than the type)

File: packages/adapter-api/src/builtins.ts [hunk 1]
```diff
@@ -39,6 +39,10 @@ export const BuiltinTypes = {
     elemID: new ElemID(GLOBAL_ADAPTER, 'json'),
```
Comment (ori-moisis) [inline packages/adapter-api/src/builtins.ts:39]: is there still a point in having a JSON builtin type now that we have `unknown`?
probably a different PR, but seems like there should be no need for it anymore

Comment (roironn) [reply packages/adapter-api/src/builtins.ts:39]: I think that JSON is more specific then unknown (for example, an unknown can be a number or a boolean, a json has to be a string) so we should probably keep it...

File: packages/workspace/src/validator.ts [hunk 2]
```diff
@@ -453,7 +454,9 @@ const validateVariableValue = (elemID: ElemID, value: Value): ValidationError[]
     return referenceValidationErrors
   }
 
-  if (!_.some(Object.values(primitiveValidators), validator => validator(value))) {
+  if (!primitiveValidators[PrimitiveTypes.STRING](value)
+    && !primitiveValidators[PrimitiveTypes.BOOLEAN](value)
+    && !primitiveValidators[PrimitiveTypes.NUMBER](value)) {
```
Comment (ori-moisis) [inline packages/workspace/src/validator.ts:459]: why was this change needed along if you also added the unknown validator?

**Review by ori-moisis — COMMENTED** [review]
new type needs to be documented in the language spec docs

File: packages/workspace/src/validator.ts (hunk 2 above)
Comment (roironn) [reply packages/workspace/src/validator.ts:459]: Its used to test the variable value - not a primitive type... (we decided to explicitly forbid objects)

Comment (ori-moisis) [reply packages/workspace/src/validator.ts:459]: we have a function called `isPrimitiveValue`, maybe it makes sense to use that here instead?
just because `unknown` is defined as a primitive, so going by type is misleading

Comment (roironn) [reply packages/workspace/src/validator.ts:459]: 👍

**Comment by ori-moisis** [comment]
code looks good, just waiting for the documentation to be added before approving

File: docs/user_guide.md [hunk 3]
```diff
@@ -463,8 +463,9 @@ The following types are supported in the language:
 | `boolean`   | boolean isOpen {}        | true / false   |
 | `json`      | json value {}            | "{ \"a\": 12 }"| A string value that expects the value to be in JSON format
 | `serviceid` | serviceid myid {}        | "ID"           | A string value that denotes an ID in the service (used by adapters to distinguish ID fields from other fields)
-| `list`      | list<string> listField {}| ["a", "b", "c"]| A list of values. contains values of a specific type
+| `list`      | "list\<string\>" listField {}| ["a", "b", "c"]| A list of values. contains values of a specific type
 | `set`       | Coming soon!             |                | Coming soon!
+| `unknown`   | unknown anyType          | anything       | a field value which is not validated, and can hold any type of data.
```
Comment (ori-moisis) [inline docs/user_guide.md:468]: nit: `anyType` sounds like a type and not a field name.
also, field definition needs `{}` (all other examples have it)
```suggestion
| `unknown`   | unknown value {}         | anything       | a field value which is not validated, and can hold any type of data.
```

**Review by ori-moisis — APPROVED** [review]
